### PR TITLE
Fix warnings, format

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Look at the first commit for the actual warnings work, the rest is just manual and automatic formatting. I tried to keep the ones that were obviously benefiting from being unformatted using `// clang-format off`, but in a file this large I may have missed some.

### CODE REVIEW

**Comment by AngheloAlf** [comment]
you know ci is angry, right?

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,23 @@
+IndentWidth: 4
+Language: Cpp
+UseTab: Never
+ColumnLimit: 120
+PointerAlignment: Left
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply .clang-format:5]: I'm used to the way all the decomps I've worked in do it, I don't really want to change it for one thing. Depends how others feel.

File: recomp.cpp [hunk 2]
```diff
@@ -35,10 +36,10 @@ using namespace std;
...
 struct Edge {
     uint32_t i;
-    uint8_t function_entry: 1;
-    uint8_t function_exit: 1;
-    uint8_t extern_function: 1;
-    uint8_t function_pointer: 1;
+    uint8_t function_entry : 1;
+    uint8_t function_exit : 1;
+    uint8_t extern_function : 1;
+    uint8_t function_pointer : 1;
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply recomp.cpp:42]: Bitfields are so rare that I basically don't care about this sort of formatting. I think it's probably better, though, it makes  the colon stand out.

File: recomp.cpp [hunk 3]
```diff
@@ -1041,18 +1129,11 @@ static uint64_t temporary_regs(void) {
...
-    TYPE_NOP,
-    TYPE_1S,
-    TYPE_2S,
-    TYPE_1D,
-    TYPE_1D_1S,
-    TYPE_1D_2S,
-    TYPE_D_LO_HI_2S,
-    TYPE_1S_POS1
-} TYPE;
+typedef enum { TYPE_NOP, TYPE_1S, TYPE_2S, TYPE_1D, TYPE_1D_1S, TYPE_1D_2S, TYPE_D_LO_HI_2S, TYPE_1S_POS1 } TYPE;
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply recomp.cpp:1135]: Well we can put comments in if you want, but this'll be gone soon anyway hopefully.

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**